src/Display/Slider/Slider.test.tsx

```diff
@@ -0,0 +1,311 @@
+import 'jest-styled-components';
+import '@testing-library/jest-dom/extend-expect';
+import { render, wait, fireEvent } from '@testing-library/react';
+import * as React from 'react';
+
+import Slider, { Props } from './Slider';
+
+type SliderComponentProp = Omit<Props, 'children'>;
+const SliderComponent = (props: SliderComponentProp) => (
+  <Slider {...props}>
+    <Slider.Item>a</Slider.Item>
+    <Slider.Item>b</Slider.Item>
+    <Slider.Item>c</Slider.Item>
+  </Slider>
+);
+
+describe('<Slider/> prop className', () => {
+  const matchSnapshot = (className: string) => {
+    it(`should match snapshot when class name ${className} is passed`, () => {
+      const { asFragment } = render(<SliderComponent className={className} />);
+      expect(asFragment()).toMatchSnapshot();
+    });
+  };
+
+  ['test', undefined].forEach(className => matchSnapshot(className));
+});
+
+describe('<Slider/> prop initialItem', () => {
+  const matchSnapshot = (initialItem: number) => {
+    it(`should match snapshot when initialItem ${initialItem} is passed`, async () => {
```

Comment:
Could you describe in more detail what's going on in this test? Why do you use `wait`? Why exactly do you need to mock `getSliderContainerDOMNode`? And why `rerender`? Some more or more elaborate comments would be helpful.

Reply:
because the slider content has a style `transformX` which is calculated based on `getSliderContainerDOMNode. getBoundingClientRect.width`, and `getBoundingClientRect` is not implemented in JSDOM, so in order to validate `transformX` is being calculated correctly, we need to mock `getSliderContainerDOMNode` to get the element width.

using `wait` is because the component do a setState in componentDidMount to set the active dot, setState is asynchronous so I need to `wait` after `render` to validate active dot position.

Since the component recalculate `transformX` in `componentWillReceiveProps`, I use rerender to trigger it.

Reply:
this is related to useLayoutEffect and useEffect, I am not super sure about this, but I think I read somewhere that if setState is executed in useEffect or componentDidMount, it's not guaranteed that setState will be executed synchronously, only useLayoutEffect guarantees synchronous setState.

Reply:
>this is related to useLayoutEffect and useEffect, I am not super sure about this, but I think I read somewhere that if setState is executed in useEffect or componentDidMount, it's not guaranteed that setState will be executed synchronously, only useLayoutEffect guarantees synchronous setState.
Also not sure but that makes sense based on my observations.

Reply:
Can you please add your explanations from above as comments in the code? This is a quite advanced test case so it would be cool if we could point to it in future and say "just look how it's done there" when we have something similar to test.